ecl/hql/hqlopt.cpp

```diff
@@ -2494,6 +2494,34 @@ IHqlExpression * CTreeOptimizer::doCreateTransformed(IHqlExpression * transforme
                 node_operator childOp = child->getOperator();
                 switch (childOp)
                 {
+                case no_selectnth:
+                    {
+                        IHqlExpression * child0 = child->queryChild(0);
+                        if (child0->getOperator()==no_newaggregate)
+                        {
+                            if (isSimpleCountExistsAggregate(child0, false, true))
+                            {
+                                IHqlExpression * selectnth = child->queryChild(1);
+                                IHqlExpression * dataset = child0->queryChild(0);
+                                if (!isShared(dataset) && getIntValue(selectnth,-1)==1 )
```

Comment:
The second half of this test could be moved earlier.  (Makes next slightly simpler).

I would think about restructuring this code so that it unwinds multiple no_addfiles if they are not shared.

i.e. count(a+b+c) -> count(a)+count(b)+count(c)